test/e2e/util.go

```diff
@@ -1749,3 +1749,105 @@ func findOVNDBLeaderPod(f *framework.Framework, cs clientset.Interface, namespac
+	return wait.PollImmediate(5*time.Second, 300*time.Second, func() (bool, error) {
+		// Check that all nodes are ready and schedulable
+		nodes, err := e2enode.GetReadySchedulableNodes(context.TODO(), f.ClientSet)
+		if err != nil {
+			framework.Logf("Error getting ready schedulable nodes: %v", err)
+			return false, nil
+		}
+
+		framework.Logf("Found %d ready schedulable nodes", len(nodes.Items))
+
+		// Check ovnkube-node pods
+		podClient := f.ClientSet.CoreV1().Pods(deploymentconfig.Get().OVNKubernetesNamespace())
+		ovnNodePods, err := podClient.List(context.Background(), metav1.ListOptions{
+			LabelSelector: "app=ovnkube-node",
+		})
+		if err != nil {
+			framework.Logf("Error listing ovnkube-node pods: %v", err)
+			return false, nil
+		}
+
+		expectedNodePods := len(nodes.Items)
+		if len(ovnNodePods.Items) != expectedNodePods {
+			framework.Logf("Expected %d ovnkube-node pods, found %d", expectedNodePods, len(ovnNodePods.Items))
+			return false, nil
+		}
+
+		// Check that all ovnkube-node pods are running
+		for _, pod := range ovnNodePods.Items {
+			if pod.Status.Phase != v1.PodRunning {
+				framework.Logf("ovnkube-node pod %s is not running (phase: %s)", pod.Name, pod.Status.Phase)
+				return false, nil
+			}
+		}
+
+		// Check ovnkube-master/control-plane pods
+		ovnMasterPods, err := podClient.List(context.Background(), metav1.ListOptions{
+			LabelSelector: "name=ovnkube-master",
+		})
+		if err != nil {
+			framework.Logf("Error listing ovnkube-master pods: %v", err)
+			return false, nil
+		}
+
+		// If no ovnkube-master pods, check for ovnkube-control-plane
+		if len(ovnMasterPods.Items) == 0 {
+			ovnMasterPods, err = podClient.List(context.Background(), metav1.ListOptions{
+				LabelSelector: "name=ovnkube-control-plane",
+			})
+			if err != nil {
+				framework.Logf("Error listing ovnkube-control-plane pods: %v", err)
+				return false, nil
+			}
+		}
+
+		if len(ovnMasterPods.Items) == 0 {
+			framework.Logf("No ovnkube-master or ovnkube-control-plane pods found")
+			return false, nil
+		}
+
+		// Check that at least one master/control-plane pod is running
+		runningMasterPods := 0
+		for _, pod := range ovnMasterPods.Items {
+			if pod.Status.Phase == v1.PodRunning {
+				runningMasterPods++
+			}
+		}
+
+		if runningMasterPods == 0 {
+			framework.Logf("No ovnkube-master/control-plane pods are running")
+			return false, nil
+		}
+
+		framework.Logf("OVN-Kubernetes cluster is healthy: %d nodes, %d ovnkube-node pods, %d running master pods",
+			len(nodes.Items), len(ovnNodePods.Items), runningMasterPods)
+		return true, nil
+	})
+}
+
+// checkBrEthIPAddress verifies that breth0 interface on the specified node has an IP address
+func checkBrEthIPAddress(nodeName string) error {
+	return wait.PollImmediate(2*time.Second, 60*time.Second, func() (bool, error) {
+		stdout, err := infraprovider.Get().ExecK8NodeCommand(nodeName, []string{"ip", "addr", "show", "breth0"})
```

Comment:
(thinking aloud) the name may be different on non-kind clusters. How does one extract bridge name in runtime?